libcaliptra/examples/generic/test.c

```diff
@@ -1179,6 +1179,78 @@ int rom_test_devid_csr(const test_info *info)
     return failure;
 }
 
+int rom_test_mldsa87_idevid_csr(const test_info *info)
+{
+    int failure = 0;
+
+    struct caliptra_buffer caliptra_idevid_csr_buf = {0};
+    caliptra_idevid_csr_buf.len = CSR_REQ_RESP_LEN;
+    // Allocte a buffer to hold the IDEV CSR using malloc
+    caliptra_idevid_csr_buf.data = malloc(caliptra_idevid_csr_buf.len);
+
+    // Check if the buffer was allocated successfully
+    if (caliptra_idevid_csr_buf.data == NULL)
+    {
+        printf("Failed to allocate memory for IDEV CSR\n");
+        return 1;
+    }
+
+    bool request_csr = true;
+    int status = boot_to_ready_for_fw(info, request_csr);
+
+    if (status)
+    {
+        dump_caliptra_error_codes();
+        failure = 1;
+    }
+
+    caliptra_wait_for_csr_ready();
+
+    int ret;
+    // Retrieve the IDEV CSR
+    if ((ret = caliptra_retrieve_idevid_csr(&caliptra_idevid_csr_buf)) != NO_ERROR)
+    {
+        printf("Failed to retrieve IDEV CSR\n");
+        printf("Error is 0x%x\n", ret);
+        failure = 1;
+    }
+    else
```

Comment:
you'd want to create an mldsa version of this and compare it to the mldsa idevid the runtime outputs